src/main/java/com/redhat/cloud/common/clowder/configsource/handlers/KafkaSaslClowderPropertyHandler.java

```diff
@@ -0,0 +1,139 @@
+    public static final String KAFKA_SSL_TRUSTSTORE_LOCATION_KEY = "kafka.ssl.truststore.location";
+    public static final String KAFKA_SSL_TRUSTSTORE_TYPE_KEY = "kafka.ssl.truststore.type";
+    public static final String CAMEL_KAFKA_SASL_JAAS_CONFIG_KEY = "camel.component.kafka.sasl-jaas-config";
+    public static final String CAMEL_KAFKA_SASL_MECHANISM_KEY = "camel.component.kafka.sasl-mechanism";
+    public static final String CAMEL_KAFKA_SECURITY_PROTOCOL_KEY = "camel.component.kafka.security-protocol";
+    public static final String CAMEL_KAFKA_SSL_TRUSTSTORE_LOCATION_KEY = "camel.component.kafka.ssl-truststore-location";
+    public static final String CAMEL_KAFKA_SSL_TRUSTSTORE_TYPE_KEY = "camel.component.kafka.ssl-truststore-type";
+    public static final String KAFKA_SSL_TRUSTSTORE_TYPE_VALUE = "PEM";
+    private static final List<String> KAFKA_SASL_KEYS = List.of(KAFKA_SASL_JAAS_CONFIG_KEY,
+            KAFKA_SASL_MECHANISM_KEY,
+            KAFKA_SECURITY_PROTOCOL_KEY,
+            KAFKA_SSL_TRUSTSTORE_LOCATION_KEY,
+            KAFKA_SSL_TRUSTSTORE_TYPE_KEY,
+            CAMEL_KAFKA_SASL_JAAS_CONFIG_KEY,
+            CAMEL_KAFKA_SASL_MECHANISM_KEY,
+            CAMEL_KAFKA_SECURITY_PROTOCOL_KEY,
+            CAMEL_KAFKA_SSL_TRUSTSTORE_LOCATION_KEY,
+            CAMEL_KAFKA_SSL_TRUSTSTORE_TYPE_KEY);
+
+    private static final List<String> KAFKA_SSL_KEYS = List.of(
+            KAFKA_SECURITY_PROTOCOL_KEY,
+            KAFKA_SSL_TRUSTSTORE_LOCATION_KEY,
+            KAFKA_SSL_TRUSTSTORE_TYPE_KEY,
+            CAMEL_KAFKA_SECURITY_PROTOCOL_KEY,
+            CAMEL_KAFKA_SSL_TRUSTSTORE_LOCATION_KEY,
+            CAMEL_KAFKA_SSL_TRUSTSTORE_TYPE_KEY);
+
+    private final boolean expose;
+    private final Optional<BrokerConfig> saslBroker;
+    private final Optional<BrokerConfig> sslBroker;
+
+    public KafkaSaslClowderPropertyHandler(ClowderConfig clowderConfig, boolean expose) {
+        super(clowderConfig);
+
+        this.expose = expose;
+        if (clowderConfig.kafka != null && clowderConfig.kafka.brokers != null) {
+            this.saslBroker = clowderConfig.kafka.brokers.stream()
+                    .filter(broker -> "sasl".equals(broker.authtype))
+                    .findAny();
+            this.sslBroker = clowderConfig.kafka.brokers.stream()
+                    .filter(broker -> "SSL".equals(broker.securityProtocol))
+                    .findAny();
+        } else {
+            this.saslBroker = Optional.empty();
+            this.sslBroker = Optional.empty();
+        }
+    }
+
+    @Override
+    public List<String> provides() {
+        if (expose) {
+            if (sslBroker.isPresent()) {
+                return KAFKA_SSL_KEYS;
+            } else {
+                return KAFKA_SASL_KEYS;
+            }
+        }
+
+        return List.of();
+    }
+
+    @Override
+    public boolean handles(String property) {
+        return KAFKA_SASL_KEYS.contains(property);
```

Comment:
Added

Reply:
Update: this change will make some tests to fail because we're not throwning an IllegalStateException anylonger. 
I've updated the handler to keep the previous logic without breaking the tests.